pyteal/ast/opup.py

```diff
@@ -0,0 +1,162 @@
+from pyteal.ast.app import OnComplete
+from pyteal.errors import TealInputError
+from pyteal.ast.while_ import While
+from pyteal.ast.expr import Expr
+from pyteal.ast.global_ import Global
+from pyteal.ast.seq import Seq
+from pyteal.ast.int import Int
+from pyteal.ast.bytes import Bytes
+from pyteal.ast.itxn import InnerTxnBuilder
+from pyteal.ast.scratchvar import ScratchVar
+from pyteal.ast.txn import TxnField, TxnType
+from pyteal.ast.for_ import For
+from pyteal.types import TealType, require_type
+from enum import Enum
+
+
+class OpUpMode(Enum):
+    """An Enum object that defines the mode used for the OpUp utility.
+
+    Note: the Explicit mode requires the app id to be provided
+    through the foreign apps array in order for it to be accessible
+    during evaluation.
+    """
+
+    """The app to call must be provided by the user."""
+    Explicit = 0
+
+    """The app to call is created then deleted for each request to increase budget."""
+    OnCall = 1
+
+
+ON_CALL_APP = Bytes("base16", "06810143")  # v6 pyteal program "Int(1)"
```

Comment:
```suggestion
ON_CALL_APP = Bytes("base16", "068101")  # v6 teal program "int 1"
```

Pyteal automatically adds a `return` op at the end of the program, but you can exclude it to save space and 1 extra compute unit (though it's kinda strange that the `return` op consumes budget in the first place 🤔 )